### PR TITLE
Base feature of the action

### PR DESCRIPTION
**Pull request by srtfisher**
Creates an action that extracts the work from @benpbolton's https://github.com/alleyinteractive/.github/blob/d691995c488fe7b6828357dd8e2be27a49b10262/.github/workflows/deploy-to-remote-repository.yml reusable workflow to a standalone action. This allows the action to be used in a existing job. 

Designed for use when deploying a `wp-content`-rooted project to a remote repository.

### CODE REVIEW

**Review by renatonascalves — APPROVED** [review]
🌵

File: .gitignore [hunk 1]
```diff
@@ -13,7 +13,7 @@ node_modules
 
 # Ignore temporary OS files
 .DS_Store
-.DS_Store?
+.DS_Store?w
```
Comment (benpbolton) [inline .gitignore]: `w` ?

File: action.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
+name: 'Deploy to Remote Repository Action'
+description: 'Uses rsync and git to deploy files/folders from a local GitHub action repository to a remote repository.'
+author: 'alleyinteractive'
+branding:
+  icon: 'cloud-lightning'
```
Comment (benpbolton) [inline action.yml:5]: like it!

File: README.md [hunk 3]
```diff
@@ -1,58 +1,44 @@
...
 
 _Notes_:
 
-- We do not leverage external actions to manage the SSH agent as we want to keep the code as simple/single-sourced as possible.
-- We must manually manage the SSH keys for the remote repository. This is typically done by adding the private key to the GitHub action secrets and then adding the public key to the remote repository (eg. as a write deploy key).
+- We do not leverage external actions to manage the SSH agent as we want to keep
+  the code as simple/single-sourced as possible.
+- We must manually manage the SSH keys for the remote repository. This is
+  typically done by adding the private key to the GitHub action secrets and then
+  adding the public key to the remote repository (eg. as a write deploy key).
```
Comment (benpbolton) [inline README.md:13]: We need a section about .deployignore behavior

**Review by benpbolton — CHANGES_REQUESTED** [review]
README should discuss `.deployignore` behavior. I made a commit with code suggestions you should review (ensuring we run the script with -e to exit on error, using a dynamic temporary directory (`$SCRATCH`) removing a spurious echo and ref. to deploy-to-vip)

This is awesome! Thanks for putting this together!

**Comment by srtfisher** [comment]
@benpbolton I added a section for it -- any chance you can re-review?

**Review by benpbolton — APPROVED** [review]
yas. great work!

**Pull request merged**